cmd/system/firecracker.go

```diff
@@ -0,0 +1,145 @@
+	gitHubLatestTemplate   = "https://api.github.com/repos/%s/releases/latest"
+	githubLatest           = "latest"
+
+	repoFlagName     = "repo"
+	versionFlagName  = "version"
+	pathFlagName     = "path"
+	progressFlagName = "progress"
+)
+
+func MakeInstallFirecracker() *cobra.Command {
+
+	command := &cobra.Command{
+		Use:   "firecracker",
+		Short: "Install Firecracker",
+		Long:  `Install Firecracker and its Jailer.`,
+		Example: `  arkade system install firecracker
+  arkade system install firecracker --version v1.0.0`,
+		SilenceUsage: true,
+	}
+
+	command.Flags().StringP(versionFlagName, "v", githubLatest, "The version for Firecracker to install")
+	command.Flags().StringP(repoFlagName, "r", "firecracker-microvm/firecracker", "The GitHub owner/repo to download Firecracker from.")
+	command.Flags().StringP(pathFlagName, "p", "/usr/local/", "Installation path, where a go subfolder will be created")
+	command.Flags().Bool(progressFlagName, true, "Show download progress")
+
+	command.RunE = func(cmd *cobra.Command, args []string) error {
+		installPath, _ := cmd.Flags().GetString(pathFlagName)
+		version, _ := cmd.Flags().GetString(versionFlagName)
+		repo, _ := cmd.Flags().GetString(repoFlagName)
+		firecrackerInstallPath := path.Join(installPath, "firecracker")
+
+		fmt.Printf("Installing Firecracker to %s\n", firecrackerInstallPath)
+
+		if err := os.MkdirAll(firecrackerInstallPath, 0755); err != nil && !os.IsExist(err) {
+			fmt.Printf("Error creating directory %s, error: %s\n", firecrackerInstallPath, err.Error())
+		}
+
+		arch, osVer := env.GetClientArch()
+
+		if strings.ToLower(osVer) != "linux" {
+			return fmt.Errorf("this app only supports Linux")
+		}
+
+		if arch != "x86_64" && arch != "aarch64" {
+			return fmt.Errorf("this app only supports x86_64 and aarch64 and not %s", arch)
+		}
+
+		if version == githubLatest {
+			v, err := getLatestGitHubRelease(repo)
+			if err != nil {
+				return err
+			}
+
+			version = v
+		} else if !strings.HasPrefix(version, "v") {
+			version = "v" + version
+		}
+
+		fmt.Printf("Installing version: %s for: %s\n", version, arch)
+
+		filename := fmt.Sprintf("firecracker-%s-%s.tgz", version, arch)
+		dlURL := fmt.Sprintf(githubDownloadTemplate, repo, version, filename)
+		fmt.Printf("Downloading from: %s\n", dlURL)
+
+		progress := true
+		outPath, err := get.DownloadFileP(dlURL, progress)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Downloaded to: %s\n", outPath)
+
+		f, err := os.OpenFile(outPath, os.O_RDONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		fmt.Printf("Unpacking Firecracker to: %s\n", firecrackerInstallPath)
+		if err := archive.Untar(f, firecrackerInstallPath, false); err != nil {
+			return err
+		}
+
+		fmt.Printf("\nexport PATH=$PATH:%s", firecrackerInstallPath)
+
+		return nil
+	}
+
+	return command
+}
+
+func getLatestGitHubRelease(repo string) (string, error) {
+
+	latestUrl := fmt.Sprintf(gitHubLatestTemplate, repo)
```

Comment:
This mechanism will be rate limited, instead we use the HTTP API and already have methods to do this that you can re-use.

Please remove and find the equivalent package from pkg/ instead.

Thanks

Reply:
Changed.